src/transform/tests/testdata/join-implementation

```diff
@@ -86,3 +84,99 @@ opt
 ----
 %0 =
 | Constant
+
+# test that JoinImplementation properly lifts MapFilterProjects
```

Comment:
Fwiw, the first test I can see what it is testing (what it says here) but the subsequent tests I really can't see what they are testing. One ends up comparing `#1 < -#1`; is it random stuff, or a specific pattern? Would these tests maybe be clearer if we invented a way to say "`opaque_predicate(#1)`" to be clear that we maybe don't care about the details? Not for this PR, but just trying to think through how to communicate intent more clearly.